From b6c8734f46b6803c4179329ca713b82ac16ed377 Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Sat, 12 Jul 2025 18:04:22 +0200
Subject: [PATCH 107/115] arm64: dts: mediatek: mt6785-xiaomi-begonia: Enable
 USB controller

This patch adds support for SuperSpeed USB, on USB C connector.
It also add support for the Extensible Host Controller Interface USB.

Signed-off-by: Dinolek <git@dinolek.me>
---
 .../mt6785-xiaomi-begonia-common.dtsi         | 48 +++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi
index 0fb5f9cd0..9df7ca060 100644
--- a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia-common.dtsi
@@ -321,6 +321,36 @@ ldo7 {
 				regulator-always-on;
 			};
 		};
+
+		tcpc {
+			compatible = "mediatek,mt6360-tcpc";
+
+			interrupts-extended = <&pio 7 IRQ_TYPE_EDGE_FALLING>;
+			interrupt-names = "PD_IRQB";
+			vbus-supply = <&otg_vbus_regulator>;
+
+			connector {
+				compatible = "usb-c-connector";
+				label = "USB-C";
+				data-role = "dual";
+				op-sink-microwatt = <10000000>;
+				power-role = "dual";
+				try-power-role = "sink";
+
+				source-pdos = <PDO_FIXED(5000, 1500,
+							 PDO_FIXED_DUAL_ROLE |
+							 PDO_FIXED_DATA_SWAP)>;
+				sink-pdos = <PDO_FIXED(5000, 2000,
+						       PDO_FIXED_DUAL_ROLE |
+						       PDO_FIXED_DATA_SWAP)>;
+
+				port {
+					typec_con_hs: endpoint {
+						remote-endpoint = <&mtu3_hs0_role_sw>;
+					};
+				};
+			};
+		};
 	};
 };
 
@@ -482,6 +512,19 @@ power {
 	};
 };
 
+&ssusb {
+	maximum-speed = "high-speed";
+	usb-role-switch;
+	vusb33-supply = <&mt6359_vusb_ldo_reg>;
+	status = "okay";
+
+	port {
+		mtu3_hs0_role_sw: endpoint {
+			remote-endpoint = <&typec_con_hs>;
+		};
+	};
+};
+
 &ufshci {
 	mediatek,ufs-disable-mcq;
 	mediatek,ufs-support-va09;
@@ -496,3 +539,8 @@ &ufshci {
 &ufsphy {
 	status = "okay";
 };
+
+&xhci {
+	vusb33-supply = <&mt6359_vusb_ldo_reg>;
+	status = "okay";
+};
-- 
2.56.0

